net-misc/gns3-{gui,server}: version bump (2.2.24) and remove old

mm1ke opened the pull request:
Hi,

This PR updates gns3 to the latest version. Some notes:
* `gns3-server` has `DISTUTILS_USE_SETUPTOOLS=rdepend` set in the ebuild and there is even a bug open that this might be incorrect. I've checked this and as far as i can see `setuptools` _is_ required at runtime. Without `setuptools` installed I get a error like following:
```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.9/gns3server", line 33, in <module>
    sys.exit(load_entry_point('gns3-server==2.2.24', 'console_scripts', 'gns3server')())
  File "/usr/lib/python-exec/python3.9/gns3server", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.9/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/usr/lib/python3.9/site-packages/gns3server/main.py", line 29, in <module>
    import gns3server.utils.get_resource
  File "/usr/lib/python3.9/site-packages/gns3server/utils/get_resource.py", line 19, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources
```
What can i do in order to avoid this warning about the setuptools usage? I would have closed this bug since the usage should be correct but i guess it would be good to make some notes somewhere?

* for `gns3-gui` i've removed `DISTUTILS_USE_SETUPTOOLS=bdepend` - this is the default anyway.
* since version 2.2.24 gns3 also supports `python3.10`, however since `dev-python/sentry-sdk` is still only at `python3.9` i can't update the ebuilds to `python3.10` yet.

mm1ke commented:
> > ```
> > * `gns3-server` has `DISTUTILS_USE_SETUPTOOLS=rdepend` set in the ebuild and there is even a bug open that this might be incorrect. I've checked this and as far as i can see `setuptools` _is_ required at runtime. Without `setuptools` installed I get a error like following:
> > ```
> 
> Thanks for checking, I have 1 machine without setuptools (uses binpkgs) and blindly removing those can be an issue.
> 
> > What can i do in order to avoid this warning about the setuptools usage? I would have closed this bug since the usage should be correct but i guess it would be good to make some notes somewhere?
> 
> Not familiar with python stuff much and what's proper, but from what I get is that upstream should be declaring this dep and being missing is why the QA check is not able to tell.
> 
> e.g. `echo setuptools >> requirements.txt` does the job for what it's worth (i.e. no more QA warning)

Thanks for the tip. :)
I've added the `setuptools` dependency in `src_prepare` and also opened a Issue upstream. (https://github.com/GNS3/gns3-server/issues/1945)

ionenwks commented:
Hm, are the gns3-server tests passing for you? No issues with 2.2.22, but these are failing with 2.2.24:
```
FAILED tests/compute/test_manager.py::test_get_abs_image_recursive_ova[pyloop] - gns3server.compute.error.ImageMi...
FAILED tests/compute/qemu/test_qemu_manager.py::test_binary_list - AssertionError: assert {'path': '/var/tmp/port...
FAILED tests/compute/qemu/test_qemu_vm.py::test_options[pyloop] - AssertionError: assert '-machine accel=tcg' == ...
FAILED tests/compute/qemu/test_qemu_vm.py::test_options_windows[pyloop] - AssertionError: assert '-machine accel=...
FAILED tests/compute/qemu/test_qemu_vm.py::test_run_with_kvm_linux_options_no_kvm[pyloop] - assert True is False
```
The test_options_windows are giving me some this-test-is-wrong vibes though.

ionenwks commented:
Hrm, actually not surprising considering CI is failing for this release upstream (same tests)

https://github.com/GNS3/gns3-server/runs/3421530738

mm1ke commented:
Hmm, seems like `test` wasn't enabled correctly on my system (i've only had the use flag `test` set but didn't enabled `FEATURES=test`). Anyway, yes it fails here too, the same tests as well..
I'm going to open another issue upstream, lets see what they can say about them.

mm1ke commented:
Updated the PR and added a patch which fixes the tests :)
As you can see from the Upstream issue, this is already fixed in master, i've just created a patch from the commit fixing this issue. With the next release i'll simple drop the patch again.

ionenwks reviewed (approved):
> Updated the PR and added a patch which fixes the tests :)

Indeed, tests are fine for me now.

> As you can see from the Upstream issue, this is already fixed in master, i've just created a patch from the commit fixing this issue. With the next release i'll simple drop the patch again.

Sounds good :)
